pkg/api/validation/validation.go

```diff
@@ -1022,11 +1023,20 @@ func ValidatePersistentVolumeClaim(pvc *api.PersistentVolumeClaim) field.ErrorLi
 func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *api.PersistentVolumeClaim) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newPvc.ObjectMeta, &oldPvc.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidatePersistentVolumeClaim(newPvc)...)
-	// if a pvc had a bound volume, we should not allow updates to resources or access modes
+	// PVController needs to update PVC.Spec w/ VolumeName. Any other change is not allowed.
+	// Claims are immutable in order to enforce quota, range limits, etc. without gaming the system.
+	if len(oldPvc.Spec.VolumeName) == 0 {
```

Comment:
I looked at PetSet and I understand what you're suggesting, but that got me wondering if I even need to do that.  Fact is, the only change allowed is volumeName from empty string to something. 

I removed the clone, added a comment about volumeName, and simply set back to "" when done.